build/ci/pr/Jenkinsfile

```diff
@@ -57,9 +60,6 @@ pipeline {
                             notOnlyDocs()
                         }
                     }
-                    agent {
-                        label 'linux'
-                    }
```

Comment:
Can you explain a bit more what is behind this change?
Why do we now need the `linux` label to build the documentation but not anymore to run E2E tests?
When do we need the `linux` label? What happens when we do not set a label, what's the default flavor?

Reply:
Sure, I should be doing it in the first place.

`Agent` block is specifying on which machine task should be executed https://jenkins.io/doc/book/pipeline/syntax/#agent. We have this block in the beginning of pipeline description, which says that PR job should run on a node marked by `linux` label.

When we have a block `parallel` which allows to run several tasks in parallel. If for this parallel tasks we will not specify any `agent`, then task will be run on the same node. Originally for unit and e2e tests task was specified `linux` label, which mean for those tasks we created 2 new nodes. And docs build runs on the same machine as original job.

https://github.com/elastic/cloud-on-k8s/pull/1351 brokes that because now we not simply build docs in Docker, but cloning 2 repos. `notOnlyDocs()` expects that it will be run in git repo, but now in docs in we have  a different structure, we have 2 repos. And it was the reason for failing launching cleanup job.

To solve this, I simply switched nodes for the job. Now e2e tests will run on the same node and for docs build a new node will be started. We can run unit/integration tests on the same node instead of e2e tests, it will be the same. 

I hope it makes my change more clear. Feel free to ask more questions if needed.

Reply:
Thank you for the explanation.